### PR TITLE
Test multiple jwt versions

### PR DESCRIPTION
**Pull request by zubin**
## Context

Currently the test suite fails when tested against JWT gem < v2. 

## Change

- Update build to test against multiple JWT gem versions (latest minor versions since 1.5.x). 
- Resolve regression against JWT < v2. 

For further details, see the new _Compatibility_ section of the README. 

Since there's no change in behaviour, there's no need to create a new version.

### CODE REVIEW

File: gemfiles/jwt_1.5.6.gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,55 @@
+PATH
```
Comment (orien) [inline gemfiles/jwt_1.5.6.gemfile.lock]: Perhaps we don't need to check in the lock files. Do we mind if gems other than `jwt` get upgraded?

**Review by orien — APPROVED** [review]
👏 🎉  Awesome! Thank you.

File: gemfiles/jwt_1.5.6.gemfile.lock (hunk 1 above)
Comment (zubin) [reply gemfiles/jwt_1.5.6.gemfile.lock]: I was thinking that too but appraisal's readme says:

> ... These are stored in the gemfiles directory, and should be added to version control to ensure that the same versions are always used.

https://github.com/thoughtbot/appraisal#under-the-hood

Unlike `Gemfile.lock`, it doesn't affect applications which depend on it.

Comment (orien) [reply gemfiles/jwt_1.5.6.gemfile.lock]: I also see [this](https://github.com/thoughtbot/appraisal#version-control).

> When using Appraisal, we recommend you check in the Gemfiles that Appraisal generates within the gemfiles directory, but exclude the lockfiles there (`*.gemfile.lock`.)

Comment (zubin) [reply gemfiles/jwt_1.5.6.gemfile.lock]: Ah ok cool. Updated: 93fe319

**Pull request merged**